src/componmnet/PointCloud/PointCloud.jsx

```diff
@@ -1,55 +1,47 @@
 import React, { Component, useEffect, useRef, useState } from 'react';
-import { main, updateLayout } from './render';
-import oriData from '../../../data/data.json';
-import { preprocess } from '../../utils';
-import { Graph } from 'igraph-wasm';
+import { main, updateCamera } from './render';
+import oriData from '../../../data/pca_3dumap_outputs_with_metadata.json';
 
 
-function PointCloud({ layout }) {
-    const [graph, setGraph] = useState(null);
+function PointCloud({ node }) {
+    const canvasRef = useRef(null);
+    const camera = useRef(null);
+    const controls = useRef(null);
+    const nodePool = getNodePool();
```

Comment:
Not sure if we want a pool that maps ids with position or a pool maps ids with Threejs Mesh (like what we have done for Network).
Position one is easy to implement.
The Mesh one will help us modify/control specific nodes by id easily.
Personally +1 for the Mesh one.